cmd_sorted_set_test.go

```diff
@@ -1416,3 +1416,139 @@ func TestSSRange(t *testing.T) {
 		equals(t, have, c.want)
 	}
 }
+
+// Test ZPOPMAX
+func TestSortedSetPopMax(t *testing.T) {
+	s, err := Run()
+	ok(t, err)
+	defer s.Close()
+	c, err := redis.Dial("tcp", s.Addr())
+	ok(t, err)
+
+	s.ZAdd("z", 1, "one")
+	s.ZAdd("z", 2, "two")
+	s.ZAdd("z", 2, "zwei")
+	s.ZAdd("z", 3, "three")
+	s.ZAdd("z", 3, "drei")
+	s.ZAdd("z", math.Inf(+1), "inf")
+
+	{
+		n, err := redis.Strings(c.Do("ZPOPMAX", "z", 2))
+		ok(t, err)
+		equals(t, 4, len(n))
+
+		equals(t, []string{"one", "1", "two", "2"}, n)
```

Comment:
I think you may have gotten this backwards.
```
redis:6379> ZRANGE z 0 -1 WITHSCORES
1) "one"
2) "1"
3) "two"
4) "2"
5) "three"
6) "3"
7) "four"
8) "4"

redis:6379> ZPOPMAX z 2
1) "four"
2) "4"
3) "three"
4) "3"
```
So you basically mixed up your ZPOPMIN and ZPOPMAX implementations.

Reply:
yes. there is a mixup.